stdlib/record.ncl

```diff
@@ -107,5 +107,93 @@
         else
           r in
         %record_insert% field r content,
+    
+    map_values: forall a b. (a -> b) -> {_: a} -> {_: b}
+    | doc m%"
+      Maps a function over every field value of a record.
+
+      For example:
+      ```nickel
+        map_values (fun x => x + 1) { hi = 2 } =>
+          { hi = 3 }
+        map (fun x => x + 1) { hello = 1, world = 2 } =>
+          { hello = 2, world = 3 }
+      ```
+    "%
+    = fun f => map (fun _field value => f value),
+
+    as_array : forall a. {_: a} -> Array {field: Str, value: a}
+    | doc m%"
+      Converts a record into an array of key-value pairs.
+
+      For example:
+      ```nickel
+        as_array { hello = "world", foo = "bar" } =>
+          [ { field = "hello", value = "world" },
+            { field = "foo", value = "bar" }
+          ]
+      ```
+    "%
+    = fun record =>
+      record
+      |> fields
+      |> array.map (fun field_ => {field = field_, value = record."%{field}"}),
+
+    from_array | forall a. Array { field: Str, value: a} -> {_: a}
+    | doc m%"
+      Convert an array of key-value pairs into a record.
+
+      For example:
+      ```nickel
+        from_array [ { field = "hello", value = "world" },
+                     { field = "foo", value = "bar" }
+                   ]
+          => { hello = "world", foo = "bar" }
+      ```
+    "%
+    = fun bindings =>
+      bindings
+      |> array.map (fun binding => { "%{binding.field}" = binding.value })
+      |> merge_array,
+
+    is_empty : forall a. {_: a} -> Bool
+    | doc m%"
+      Check whether a record is empty.
+      
+      ```nickel
+      is_empty {} => true
+      is_empty { foo = 1 } => false
+      ```
+    "%
+    = (==) {},
+
+    merge_array | forall a. Array {_: a} -> {_: a}
+    | doc m%"
+      Merge an array of records.
+
+      For example:
+      ```nickel
+      merge_array [ { foo = 1 }, { bar = 2 } ]
+        => { foo = 1, bar = 2 }
+      ```
+    "%
+    = array.foldl (fun acc r => acc & r) {},
+
+    filter | (Str -> Dyn -> Bool) -> {_: Dyn} -> {_: Dyn}
```

Comment:
Ditto: can't this be statically typed?

Reply:
The implementation used destructuring and needed @matthew-healy's #1099 to typecheck. But since that's merged now, it was just a rebase away 🎉

Reply:
Ah, I see :slightly_smiling_face: